scipy/sparse/linalg/_eigen/arpack/tests/test_arpack.py

```diff
@@ -87,38 +87,41 @@ def _get_test_tolerance(type_char, mattype=None, D_type=None, which=None):
 
 
 def generate_matrix(N, complex_=False, hermitian=False,
-                    pos_definite=False, sparse=False):
-    M = np.random.random((N, N))
+                    pos_definite=False, sparse=False, rng=None):
+    if rng is None:
+        rng = np.random
```

Comment:
Did you mean to write `np.random` (keeps using the global functions) or `np.random.default_rng` (or `RandomState`)?